### PR TITLE
Protectwise - upgraded integration and deprecated scripts

### PR DESCRIPTION
**Pull request by liorkol**
fixes https://github.com/demisto/server/issues/6457

### CODE REVIEW

File: Integrations/integration-ProtectWise.yml [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
 description: Cloud based Security Network DVR
-detaileddescription: |-
-  Integration with ProtectWise cloud product.
-  Please note you can use either token (acquired by api call : https://api.protectwise.com/momfodhxhz/#access-token-token) or user email/password for authentication.
+detaileddescription: Integration with ProtectWise cloud product.
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: Did you deliberately remove all the detailed description? if adding back make sure to use proper markdown syntax for URLs

File: Integrations/integration-ProtectWise.yml [hunk 2]
```diff
@@ -63,96 +98,45 @@ script:
...
+            params.insecure,
+            params.proxy
             );
         if (res.StatusCode !== 200) {
-            throw 'Failed to execute download Event Pcap request:' + res.StatusCode + ', body: ' + res.Body;
+            throw 'Failed to execute ' + url + ' : ' + res.StatusCode + ', body: ' + res.Body;
         }
-        return saveFile(res.Body);
+
+        return (raw ? res.Body : JSON.parse(res.Body));
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: You should wrap parse with try catch, so we can return proper error if body was not a proper json
Look at other places you do parse as well

File: Integrations/integration-ProtectWise.yml [hunk 3]
```diff
@@ -201,101 +187,300 @@ script:
...
+        for (var i = 0; i<keys.length; i++) {
+            val = event[keys[i]];
+            if (TIME_FIELDS.indexOf(keys[i]) > -1) {
+                val = convertTimestampToString(val);
+            }
+            labels.push({'type': keys[i], 'value': String(val)});
+        }
+        return {
+            "name": event.message,
+            "labels": labels,
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: Need to add rawJSON so it will be able to  go through mapping properly

File: Integrations/integration-ProtectWise.yml [hunk 4]
```diff
@@ -201,101 +187,300 @@ script:
...
     switch (command) {
         case 'test-module':
-            var token = getToken();
            if (token && token.length > 0){
                 return true;
             }
             return false;
+        case 'fetch-incidents':
+            var lastRun = getLastRun();
+            var startTime = (lastRun && lastRun.startTime ? lastRun.startTime : (new Date().getTime()- 1000 * 60 ));
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: This is last minute back, usually we give it a but more

File: Integrations/integration-ProtectWise.yml [hunk 5]
```diff
@@ -201,101 +187,300 @@ script:
...
+                        }
+                    }
+                } else {
+                    skip = false;
+                }
+                if(!skip) {
+                    incidents.push(createIncidentFromEvent(res[i]));
+                }
+            }
+            setLastRun({startTime: startTime});
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: You are not incrementing the lastRun, or i am missing something

File: Integrations/integration-ProtectWise.yml [hunk 6]
```diff
@@ -201,101 +187,300 @@ script:
...
+            return JSON.stringify(incidents);
         case 'sensors':
-            var token = getToken();
-            return {Type: 1, Contents: sensors(args.sensorId,token), ContentsFormat: 'json'};
+        case 'protectwise-show-sensors':
+            var url = params.url+'sensors';
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: In most integrations we protect ourselves from cases placed / didnt place the `/`

File: Integrations/integration-ProtectWise.yml [hunk 7]
```diff
@@ -201,101 +187,300 @@ script:
...
+                   }
+               });
+            });
+
+            return {
+                Type: entryTypes.note,
+                Contents: raw,
+                ContentsFormat: formats.json,
+                HumanReadable: tableToMarkdown('Protectwise Event Search', items, eventHeaders),
+                EntryContext: {'Protectwise.Event(val.id == obj.id)': items}
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: Dont you have here a lot of events ? what ID are you referring to ?

File: Integrations/integration-ProtectWise.yml [hunk 8]
```diff
@@ -201,101 +187,300 @@ script:
...
+                   }
+               });
+            });
+
+            return {
+                Type: entryTypes.note,
+                Contents: raw,
+                ContentsFormat: formats.json,
+                HumanReadable: tableToMarkdown('Protectwise Observation Search', items, headers),
+                EntryContext: {'Protectwise.Observation(val.id == obj.id)': items}
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: look at the comment above

File: Integrations/integration-ProtectWise.yml [hunk 9]
```diff
@@ -201,101 +187,300 @@ script:
...
+                ContentsFormat: formats.json,
+                HumanReadable: md,
+                EntryContext: {'Protectwise.Event(val.id == obj.id)': {'id': res.id, 'PCAPSize': res.estimatedSize}}
+            };
         case 'observation-pcap-download':
-            var token = getToken();
-            return {Type: 3, FileID: observationEventPcap(args.id,args.sensorId,args.filename,token), File: args.filename, Contents: 'we must have contents for an entry'};
+        case 'protectwise-observation-pcap-download':
+            var filename = (args.filename && args.filename.length > 0) ? args.filename : ( args.sensorId+'-'+args.id + '.pcap');
+            var url = params.url+'pcaps/observations/'+args.sensorId+'/'+args.id + '?filename=' + filename;
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: you need to escape filename (we have a function for that in common

File: Integrations/integration-ProtectWise.yml [hunk 10]
```diff
@@ -201,101 +187,300 @@ script:
...
+            return {
+                Type: entryTypes.note,
+                Contents: res,
+                ContentsFormat: formats.json,
+                HumanReadable: md,
+                EntryContext: {'Protectwise.Observation(val.id == obj.id && val.sensorId == obj.sensorId)': {'id': args.id, 'sensorId': args.sensorId, 'PCAPSize': res.estimatedSize}}
+            };
         case 'get-token':
-            return getToken();
+            return token;
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: above you used the function getToken
why did you remove it here ?

File: Integrations/integration-ProtectWise.yml (hunk 1 above)
Comment (liorkol) [inline Integrations/integration-ProtectWise.yml]: Removed the URL. Token no longer an option to use instead of password.

File: Integrations/integration-ProtectWise.yml (hunk 4 above)
Comment (liorkol) [inline Integrations/integration-ProtectWise.yml]: Updated to 5mins

File: Integrations/integration-ProtectWise.yml (hunk 6 above)
Comment (liorkol) [inline Integrations/integration-ProtectWise.yml]: Good idea. Done

File: Integrations/integration-ProtectWise.yml (hunk 7 above)
Comment (liorkol) [inline Integrations/integration-ProtectWise.yml]: Right.. it's an array of events. But it merges on a per-item basis
Seemed to work at least when I tested, let me know if you think I am misunderstanding

File: Integrations/integration-ProtectWise.yml (hunk 10 above)
Comment (liorkol) [inline Integrations/integration-ProtectWise.yml]: getToken always gets called, before checking which command it is - so no reason to call it again, just return the token we already got

File: Integrations/integration-ProtectWise.yml (hunk 9 above)
Comment (liorkol) [inline Integrations/integration-ProtectWise.yml]: Done

**Comment by liorkol** [comment]
@aviadl Fixed and tested, merge when ready

File: Integrations/integration-ProtectWise.yml [hunk 11]
```diff
@@ -230,10 +239,19 @@ script:
...
         case 'fetch-incidents':
             var lastRun = getLastRun();
-            var startTime = (lastRun && lastRun.startTime ? lastRun.startTime : (new Date().getTime()- 1000 * 60 ));
-            var data = eventsSearch(startTime , new Date().getTime() , params.eventType, params.killChainStage, params.threatLevel,
+            var now = (new Date()).getTime();
+            if (!lastRun || !lastRun.time) {
+                // First time, retrieve events from the last 10 min
+                lastRun = {time: now - 10 * 60 * 1000};
+            }
+            setLastRun({time: now});
```
Comment (aviadl) [inline Integrations/integration-ProtectWise.yml]: You should set last run, only if fetch was successful
Else you may lose timeframes
If you intend to have it this way, then ignore
